helpers/exports.js

```diff
@@ -35,3 +39,76 @@ exports.resendVerificationEmail = async (req, res) => {
 
   }
 };
+
+/**
+  * @function searchByTagAuthorOrTitle
+  * @summary Articles can be searched by Tag, Author or TItle
+  * @param {object} req - Request object
+  * @param {object} res - Response object
+  * @returns {object} returns the list of articles matching the search criteria
+  */
+exports.searchByTagAuthorOrTitle = (req, res) => {
+  const searchParameters = {};
+  // Get every parameter and key passed in the query string
+  const keysAndValues = Object.entries(req.query);
+  // Assign page number as the first page if the page number is not passed
+  let pageNumber = typeof req.query.pageNumber === 'undefined' ? 1 : req.query.pageNumber;
+  let pageSize = typeof req.query.pageSize === 'undefined' ? 10 : req.query.pageSize;
```

Comment:
I'll suggest you use object destructuring for the `req.query` object and in addition, you can  change the logic to be `let { pageNumber = 1, pageSize = 10 } = req.query;`   combining object destructuring with default value, in case `pageNumber or pageSize` is `undefined`, the default value will be assigned. It will reduce the number of codes and improve readability, you can try it.

Reply:
Ok. I'd check this out.

Reply:
@ascii-dev Yes that was a solution I discussed with some team members who had implemented the article section of the code. The methodology used was somewhat similar to this style here, so i decided to be consistent in that regard. I also thought that making an article belong to multiple categories using relationships will generate more models (tables)with more relationships. This will be expensive in code and query. So i decided to use a less expensive approach here. Cheers. Feel free to comment.